### ISSUE TITLE
cannot import name 'sphere_synt' from partially initialized module 'ref'

### ISSUE THREAD

**Issue by XiaoHaoPan** [THU-DA-6D-Pose-Group/self6dpp#16]
When I run det/yolov4/train_yolov4.sh configs/yolov4/bop_pbr/yolov4x_640_augCosyAAEGray_ranger_lm_pbr_lm_test.py 0, It produces an error. I found that there are no “sphere_synt.py” files under the “def” directory.How can I fix it?

**Comment by wangg12** [comment]
It is not needed for this repo. You can just comment it out.

**Comment by XiaoHaoPan** [comment]
Checkpoint pretrained_models/yolov4/yolov4x-mish.pth not found!Now I have this problem.Please how can I solve it?

**Comment by wangg12** [comment]
https://github.com/WongKinYiu/PyTorch_YOLOv4

**Comment by XiaoHaoPan** [comment]
How do I use it？I click on the "weights" of the pre-trained model inside, but it shows that the file does not exist.

**Comment by XiaoHaoPan** [comment]
Prior to this,when I run sh scripts/compile. sh command, it displays the following content.

![image](https://user-images.githubusercontent.com/127167436/232410665-ae417bee-7e88-4b37-8102-c31f17d04cdf.png)
May I ask if this has any impact?

**Comment by wangg12** [comment]
Looks like the egl renderer could not be successfully built in your env. You need to fix the problem.

**Comment by XiaoHaoPan** [comment]
Yes, how can I fix this problem？
![image](https://user-images.githubusercontent.com/127167436/232420296-9cd8fd71-2c2c-4ac1-809f-8ee5941824eb.png)
My env has already been installed PyOpenGL
![image](https://user-images.githubusercontent.com/127167436/232421024-b31f2904-cd79-4953-ac48-33048a911438.png)

**Comment by wangg12** [comment]
Maybe you can refer to: https://github.com/shanice-l/gdrnpp_bop2022/issues/35 and some other related issues online.

**Comment by XiaoHaoPan** [comment]
Thank you very much for your help! I have resolved the egl renderer issue, but this(https://github.com/THU-DA-6D-Pose-Group/self6dpp/issues/16#issuecomment-1510791611) still exists.
![image](https://user-images.githubusercontent.com/127167436/232433015-59df3c73-43ec-4a3e-9b8e-8ed31a0d76ba.png)
I entered this(https://github.com/WongKinYiu/PyTorch_YOLOv4), but its weights file no longer exists

**Comment by XiaoHaoPan** [comment]
Are these different profiles different datasets? What is the difference between config inside the _base_ folder and the four outside?

![image](https://user-images.githubusercontent.com/127167436/232645010-7efcc9c5-928a-400e-91c4-16acf5c63455.png)

Yolov4's processing on LM I see only test.py. So is the reason for my error(https://github.com/THU-DA-6D-Pose-Group/self6dpp/issues/16#issuecomment-1510791611) being because I run det/yolov4/train_yolov4.sh and let configs set the yolov4x_640_augCosyAAEGray_ranger_lm_pbr_lm_test?

![image](https://user-images.githubusercontent.com/127167436/232645511-b903fe5d-d410-4d68-ba28-5e86f238b0f2.png)

**Comment by wangg12** [comment]
Please find the answer in the code https://github.com/THU-DA-6D-Pose-Group/self6dpp/blob/main/det/yolov4/main_yolov4.py

**Comment by XiaoHaoPan** [comment]
This looks like the following code is causing problems, but I don't know how to properly modify it to make it correct
https://github.com/THU-DA-6D-Pose-Group/self6dpp/blob/d94eaf7045ff3f2d992f23735e7dc3011338a3fd/det/yolov4/main_yolov4.py#L181

![JVMWN%MUSQOXX4U22R1F(`A](https://user-images.githubusercontent.com/127167436/232697449-bebc595b-bf31-4708-95c5-442c0cfdc723.png)

![Z {ATZ17 9D9VH8`RC5PTSQ](https://user-images.githubusercontent.com/127167436/232697785-b545767c-cf0c-4d44-9bbf-17011ffe53d1.png)

**Comment by XiaoHaoPan** [comment]
Can you provide a pretrained_ models of yolov4x-mish.pth? Because I found that its download link has expired in its Yolov4 GitHub.

**Comment by wangg12** [comment]
I don't have the yolov4 models now. Please contact the original authors.

**Comment by XiaoHaoPan** [comment]
Thanks to your help, I am now able to train The Detector.

Can I adjust the TOTAL_EPOCHS=4 here to make it less training? Because I find that training the detector takes too long, I use a piece of 2080ti, will this have an impact beyond accuracy?

![image](https://user-images.githubusercontent.com/127167436/232977772-e0f275d4-38a0-472b-87bd-fc16b0b119a5.png)

**Comment by wangg12** [comment]
You can try it and see how much it impacts the accuracy.
